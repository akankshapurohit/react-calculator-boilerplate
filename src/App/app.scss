.DCMN-logo {
    width: 100%;
    height: 100vh;
    display: flex;
    justify-content: center;

    img {
        width: 300px;
    }
}
