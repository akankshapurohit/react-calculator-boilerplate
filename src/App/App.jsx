import React from 'react';
import './app.scss';

class App extends React.PureComponent {
    render() {
        return (
            <div className="DCMN-logo">
                <img src="https://dcmn.com/assets/dcmn-logo-redesign-black.svg" alt="DCMN logo" />
            </div>
        );
    }
}

export default App;
