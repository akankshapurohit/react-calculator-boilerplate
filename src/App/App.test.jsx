import React from 'react';
import { shallow } from 'enzyme';

import App from './App';

describe('App', () => {
    let consoleSpy;

    beforeAll(() => {
        consoleSpy = jest.spyOn(console, 'error');
    });

    it('renders the App component and matches snapshot', () => {
        const component = shallow(<App />);

        expect(component).toMatchSnapshot();
    });

    it('renders logo', () => {
        const component = shallow(<App />);

        expect(component.find('img')).toHaveLength(1);
    });

    it('shows no errors on console', () => {
        expect(consoleSpy).not.toHaveBeenCalled();
    });
});
