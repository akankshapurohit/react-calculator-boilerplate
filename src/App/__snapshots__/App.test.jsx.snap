// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the App component and matches snapshot 1`] = `
<div
  className="DCMN-logo"
>
  <img
    alt="DCMN logo"
    src="https://dcmn.com/assets/dcmn-logo-redesign-black.svg"
  />
</div>
`;
